runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingPortablePipelineTranslator.java

```diff
@@ -423,8 +432,133 @@ private void translateImpulse(
           String id,
           RunnerApi.Pipeline pipeline,
           StreamingTranslationContext context) {
+    // TODO: Fail on stateful DoFns for now.
+    // TODO: Support stateful DoFns by inserting group-by-keys where necessary.
+    // TODO: Fail on splittable DoFns.
+    // TODO: Special-case single outputs to avoid multiplexing PCollections.
+    RunnerApi.Components components = pipeline.getComponents();
+    RunnerApi.PTransform transform = components.getTransformsOrThrow(id);
+    Map<String, String> outputs = transform.getOutputsMap();
+    RehydratedComponents rehydratedComponents =
+            RehydratedComponents.forComponents(components);
+
+    BiMap<String, Integer> outputMap =
+            FlinkPipelineTranslatorUtils.createOutputMap(outputs.keySet());
+    Map<String, Coder<WindowedValue<?>>> outputCoders = Maps.newHashMap();
+    for (String localOutputName : new TreeMap<>(outputMap.inverse()).values()) {
+      String collectionId = outputs.get(localOutputName);
+      Coder<WindowedValue<?>> windowCoder = (Coder) instantiateCoder(collectionId, components);
+      outputCoders.put(localOutputName, windowCoder);
+    }
+
+    final RunnerApi.ExecutableStagePayload stagePayload;
+    try {
+      stagePayload = RunnerApi.ExecutableStagePayload.parseFrom(transform.getSpec().getPayload());
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    String inputPCollectionId =
+            Iterables.getOnlyElement(transform.getInputsMap().values());
 
-    throw new RuntimeException("executable stage translation not implemented");
+    // TODO: is this still relevant?
+    // we assume that the transformation does not change the windowing strategy.
+    RunnerApi.WindowingStrategy windowingStrategyProto =
```

Comment:
Removed, it was a remnant of translation code from old runner (only used in case of stateful ParDo). Since key handling and output encoding will happen in the SDK harness, we probably won't need to know the windowing strategy in the operator.